Stp.TestingApi/Controllers/ExerciseController.cs

```diff
@@ -0,0 +1,185 @@
+        {
+            Exercise exercise = FindExercise(exerciseId);
+
+            if (exercise == null)
+            {
+                return BadRequest();
+            }
+
+            exercise.Name = name;
+            _db.SaveChanges();
+
+            return Ok();
+        }
+
+        [HttpPut(nameof(UpdateExerciseDuration))]
+        /* https://localhost:5001/api/Exercise/UpdateExerciseDuration */
+        public IActionResult UpdateExerciseDuration(long exerciseId, [FromBody]int duration)
+        {
+            Exercise exercise = FindExercise(exerciseId);
+
+            if (exercise == null)
+            {
+                return BadRequest();
+            }
+
+            exercise.DurationMinutes = duration;
+            _db.SaveChanges();
+
+            return Ok();
+        }
+
+        [HttpPut(nameof(UpdateExercisePoints))]
+        /* https://localhost:5001/api/Exercise/UpdateExercisePoints */
+        public IActionResult UpdateExercisePoints(long exerciseId, [FromBody]int points)
+        {
+            Exercise exercise = FindExercise(exerciseId);
+
+            if (exercise == null)
+            {
+                return BadRequest();
+            }
+
+            exercise.Points = points;
+            _db.SaveChanges();
+
+            return Ok();
+        }
+
+        [HttpPut(nameof(UpdateExerciseComplexity))]
+        /* https://localhost:5001/api/Exercise/UpdateExerciseComplexity */
+        public IActionResult UpdateExerciseComplexity(long exerciseId, [FromBody]ExerciseComplexity complexity)
+        {
+            Exercise exercise = FindExercise(exerciseId);
+
+            if (exercise == null)
+            {
+                return BadRequest();
+            }
+
+            exercise.Complexity = (int)complexity;
+            _db.SaveChanges();
+
+            return Ok();
+        }
+
+        [HttpPut(nameof(UpdateExerciseDescription))]
+        /* https://localhost:5001/api/Exercise/UpdateExerciseDescription */
+        public IActionResult UpdateExerciseDescription(long exerciseId, [FromBody]string description)
+        {
+            Exercise exercise = FindExercise(exerciseId);
+
+            if (exercise == null)
+            {
+                return BadRequest();
+            }
+
+            exercise.Description = description;
+            _db.SaveChanges();
+
+            return Ok();
+        }
+
+        [HttpPut(nameof(DeleteExercise))]
+        /* https://localhost:5001/api/Exercise/DeleteExercise */
+        public IActionResult DeleteExercise(long exerciseId)
+        {
+            Exercise exercise = FindExercise(exerciseId);
+
+            if (exercise == null)
+            {
+                return BadRequest();
+            }
+
+            exercise.IsDeleted = true;
+            _db.SaveChanges();
+
+            return Ok();
+        }
+
+        private Exercise FindExercise(long id)
+        {
+            var res = _db.ExerciseList
```

Comment:
It's OK, but when finding by PK, it's better to use Find function, which can be more effective then FirstOrDefault in some cases.